packages/assets-controllers/src/crypto-compare-service/crypto-compare.ts

```diff
@@ -0,0 +1,154 @@
+  currentCurrency: string,
+  nativeCurrency: string,
+  includeUSDRate?: boolean,
+) {
+  nativeCurrency = nativeCurrency.toUpperCase();
+  const fsym = nativeSymbolOverrides.get(nativeCurrency) ?? nativeCurrency;
+  return (
+    `${CRYPTO_COMPARE_DOMAIN}/data/price?fsym=` +
+    `${fsym}&tsyms=${currentCurrency.toUpperCase()}` +
+    `${includeUSDRate && currentCurrency.toUpperCase() !== 'USD' ? ',USD' : ''}`
+  );
+}
+
+/**
+ * Get the CryptoCompare API URL for getting the conversion rate from a given array of native currencies
+ * to the given currency. Optionally, the conversion rate from the native currency to USD can also be
+ * included in the response.
+ *
+ * @param fsyms - The native currencies to get conversion rates for.
+ * @param tsyms - The currency to convert to.
+ * @param includeUSDRate - Whether or not the native currency to USD conversion rate should be included.
+ * @returns The API URL for getting the conversion rates.
+ */
+function getMultiPricingURL(
+  fsyms: string,
+  tsyms: string,
+  includeUSDRate = false,
+) {
+  const updatedTsyms =
+    includeUSDRate && !tsyms.includes('USD') ? `${tsyms},USD` : tsyms;
+
+  const params = new URLSearchParams();
+  params.append('fsyms', fsyms);
+  params.append('tsyms', updatedTsyms);
+
+  const url = new URL(`${CRYPTO_COMPARE_DOMAIN}/data/pricemulti`);
+  url.search = params.toString();
+
+  return url.toString();
+}
+
+/**
+ * Handles an error response from the CryptoCompare API.
+ * Expected error response format
+ * { Response: "Error", Message: "...", HasWarning: false }
+ *
+ * @param json - The JSON response from the CryptoCompare API.
+ * @param json.Response - The response status.
+ * @param json.Message - The error message.
+ */
+function handleErrorResponse(json: { Response?: string; Message?: string }) {
+  if (json.Response === 'Error') {
+    throw new Error(json.Message);
+  }
+}
+
+/**
+ * Fetches the exchange rate for a given currency.
+ *
+ * @param currency - ISO 4217 currency code.
+ * @param nativeCurrency - Symbol for base asset.
+ * @param includeUSDRate - Whether to add the USD rate to the fetch.
+ * @returns Promise resolving to exchange rate for given currency.
+ */
+export async function fetchExchangeRate(
+  currency: string,
+  nativeCurrency: string,
+  includeUSDRate?: boolean,
+): Promise<{
+  conversionRate: number;
+  usdConversionRate: number;
+}> {
+  const json = await handleFetch(
+    getPricingURL(currency, nativeCurrency, includeUSDRate),
+  );
+
+  handleErrorResponse(json);
+  const conversionRate = Number(json[currency.toUpperCase()]);
+
+  const usdConversionRate = Number(json.USD);
+  if (!Number.isFinite(conversionRate)) {
+    throw new Error(
+      `Invalid response for ${currency.toUpperCase()}: ${
+        json[currency.toUpperCase()]
+      }`,
+    );
+  }
+
+  if (includeUSDRate && !Number.isFinite(usdConversionRate)) {
+    throw new Error(`Invalid response for usdConversionRate: ${json.USD}`);
+  }
```

Comment:
Just moving "conversion rate" value closer to their checks makes more sense, WDYT?

```suggestion
  
  const conversionRate = Number(json[currency.toUpperCase()]);
  if (!Number.isFinite(conversionRate)) {
    throw new Error(
      `Invalid response for ${currency.toUpperCase()}: ${
        json[currency.toUpperCase()]
      }`,
    );
  }

  const usdConversionRate = Number(json.USD);
  if (includeUSDRate && !Number.isFinite(usdConversionRate)) {
    throw new Error(`Invalid response for usdConversionRate: ${json.USD}`);
  }
```

Reply:
This is part of the method that was here previously, it's not part of this PR. I'll try to move the file again to see if gh reflects it properly